src/alire/alire-origins-deployers.ads

```diff
@@ -63,6 +63,16 @@ package Alire.Origins.Deployers is
    --  case, it is polite not to uninstall anything installed in the user
    --  system.
 
+   function Compute_Hash (This   : Deployer;
+                          Folder : String;
+                          Kind   : Hashes.Kinds) return Hashes.Any_Digest with
+     Pre'Class =>
+       Supports_Hashing (This.Base.Kind) or else raise Program_Error;
```

Comment:
I don't thin you have to raise a Program_Error here, if the precondition is false an exception will be raised anyway.

Reply:
It was to ensure it is a Program_Error and not an assertion failed.